src/crate/client/cursor.py

```diff
@@ -238,3 +242,64 @@ def get_default_converter() -> Converter:
         Return the standard converter instance.
         """
         return DefaultTypeConverter()
+
+    @property
+    def time_zone(self):
+        """
+        Get the current time zone.
+        """
+        return self._time_zone
+
+    @time_zone.setter
+    def time_zone(self, tz):
+        """
+        Set the time zone.
+
+        It supports different ways to populate. Some examples::
+
+        - ``datetime.timezone.utc``
+        - ``datetime.timezone(datetime.timedelta(hours=7), name="MST")``
+        - ``pytz.timezone("Australia/Sydney")``
+        - ``+0530`` (UTC offset in string format)
+        """
+
+        # Do nothing when time zone is reset.
+        if tz is None:
+            self._time_zone = None
+            return
+
+        # Requesting datetime-aware `datetime` objects needs the data type converter.
+        # Implicitly create one, when needed.
+        if self._converter is None:
+            self._converter = Converter()
+
+        # When the time zone is given as a string, assume UTC offset format, e.g. `+0530`.
+        if isinstance(tz, str):
+            tz = self._timezone_from_utc_offset(tz)
+
+        self._time_zone = tz
+
+        def _to_datetime_with_tz(value: t.Optional[float]) -> t.Optional[datetime]:
+            """
+            Convert CrateDB's `TIMESTAMP` value to a native Python `datetime`
+            object, with timezone-awareness.
+            """
+            if value is None:
+                return None
+            return datetime.fromtimestamp(value / 1e3, tz=self._time_zone)
+
+        # Register converter function for `TIMESTAMP` type.
+        self._converter.set(CrateDatatypeIdentifier.TIMESTAMP, _to_datetime_with_tz)
+
+    @staticmethod
+    def _timezone_from_utc_offset(tz) -> timezone:
+        """
+        Convert UTC offset in string format (e.g. `+0530`) into `datetime.timezone` object.
+        """
+        assert len(tz) == 5, f"Time zone '{tz}' is given in invalid UTC offset format"
+        try:
+            hours = int(tz[:3])
+            minutes = int(tz[0] + tz[3:])
+            return timezone(timedelta(hours=hours, minutes=minutes), name=tz)
+        except Exception as ex:
+            raise ValueError(f"Time zone '{tz}' is given in invalid UTC offset format: {ex}")
```

Comment:
I wonder if this helper function should gracefully also support the `+05:30` format, like it is used in the example for the [MySQL Connector/Python](https://github.com/mysql/mysql-connector-python) driver [^1].

[^1]: https://dev.mysql.com/doc/connector-python/en/connector-python-api-mysqlconnection-time-zone.html

Reply:
Imho, I wouldn't bother to expand support to other possible formats, I think that documenting the acceptable formats is enough.

Reply:
It is now documented here:

> For the time_zone keyword argument, different data types are supported. The available options are:

    [...]
    +0530 (UTC offset in string format)

-- https://github.com/crate/crate-python/blob/98630d0167b0f7802586fb7d3ffd7f25087d27ba/docs/query.rst#timestamp-conversion-with-time-zone